spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/IcebergSparkSqlExtensionsParser.scala

```diff
@@ -54,8 +66,13 @@ class IcebergSparkSqlExtensionsParser(delegate: ParserInterface) extends ParserI
   /**
    * Parse a string to a raw DataType without CHAR/VARCHAR replacement.
    */
-  override def parseRawDataType(sqlText: String): DataType = {
-    delegate.parseRawDataType(sqlText)
+  def parseRawDataType(sqlText: String): DataType = {
```

Comment:
I don't think that Iceberg actually uses this method anywhere, so I would opt for a simpler approach: either throw `UnsupportedOperationException` or return `parseDataType(sqlText)`.

Reply:
In that case, I'll just throw `UnsupportedOperationException` here.